File: OfflineFetcher/app/src/main/java/de/jbamberger/offlinefetcher/source/reddit/RedditPostDao.java
package de.jbamberger.offlinefetcher.source.reddit;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import static android.arch.persistence.room.OnConflictStrategy.REPLACE;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Dao
public interface RedditPostDao {

    @Insert(onConflict = REPLACE)
    void save(RedditPost post);

    @Query("SELECT * FROM redditpost WHERE id = :subreddit")
    LiveData<RedditPost> load(String subreddit);

}

File: OfflineClient/cached-api/src/main/java/de/jbamberger/api/provider/reddit/RedditPostDao.kt
package de.jbamberger.api.provider.reddit

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy.REPLACE
import android.arch.persistence.room.Query

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Dao
interface RedditPostDao {

    @Insert(onConflict = REPLACE)
    fun save(post: RedditPost)

    @Query("SELECT * FROM redditpost WHERE id = :subreddit")
    fun load(subreddit: String): LiveData<RedditPost>

}
